test/addons_test.go

```diff
@@ -366,24 +574,37 @@ func testgroup(t *testing.T, groupname string, version string, jobs ...clusterTe
 			return err
 		}
 
-		t.Logf("found old version of addon %s %s (revision %s) and new version %s (revision %s)", newAddon.GetName(), oldRev, oldVersion, newVersion, newRev)
-		if oldVersion.GT(newVersion) {
-			return fmt.Errorf("revisions for addon %s are broken, previous revision %s is newer than current %s", newAddon.GetName(), oldVersion, newVersion)
-		}
-		if !newVersion.GT(oldVersion) {
+		if newVersion.EQ(oldVersion) {
 			t.Logf("skipping upgrade test for addon %s, it has not been updated", newAddon.GetName())
 			continue
+		} else if oldVersion.GT(newVersion) {
+			return fmt.Errorf("revisions for addon %s are broken, previous revision %s is newer than current %s", newAddon.GetName(), oldVersion, newVersion)
+		} else {
+			t.Logf("found old version of addon %s %s (revision %s) and new version %s (revision %s)", newAddon.GetName(), oldRev, oldVersion, newVersion, newRev)
 		}
 
 		t.Logf("INFO: addon %s was modified and will be upgrade tested", newAddon.GetName())
+
 		addonUpgrade, err := addontesters.UpgradeAddon(t, tcluster, oldAddon, newAddon)
 		if err != nil {
 			return err
 		}
 
+		// append for upgrades
 		addonUpgrades = append(addonUpgrades, addonUpgrade)
 	}
 
+	addonDeployment, err := addontesters.DeployAddons(t, tcluster, deployments...)
+	if err != nil {
+		return err
+	}
+
+	if len(addonUpgrades) == 0 {
+		t.Logf("INFO: NO UPGRADES to be tested, EXITING")
+		close(stop)
+		return nil
+	}
+
 	th := testharness.NewSimpleTestHarness(t)
 	th.Load(
 		addontesters.ValidateAddons(addons...),
```

Comment:
Can't comment on line 612 but below with `addonDefaults` plan (terribly named 🙄 maybe we should rename it to `waitForAddons` or something clearer lol) - This is waiting for all the addons including addons that haven't been deployed yet (ones that will be upgrade-tested, as they are deployed in the Upgrade plan) which is why we're seeing the tests fail in #817 . I think we should be passing in `deployments...` instead. `addonDefaults` is defined at [line 543](https://github.com/mesosphere/kubernetes-base-addons/pull/809/files#diff-6b1d50a89d4ac455215d0f0969ed4e5507e3a91528e4ef6a20e37d6d515781b2R543) (have to expand the diff to see)